src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/WasmBrowserTestRunner.cs

```diff
@@ -0,0 +1,190 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Server.Features;
+using Microsoft.DotNet.XHarness.CLI.CommandArguments.Wasm;
+using Microsoft.DotNet.XHarness.Common.CLI;
+using Microsoft.Extensions.Logging;
+
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+
+using SLogLevel = OpenQA.Selenium.LogLevel;
+using LogLevel = Microsoft.Extensions.Logging.LogLevel;
+
+namespace Microsoft.DotNet.XHarness.CLI.Commands.Wasm
+{
+    internal class WasmBrowserTestRunner
+    {
+        private readonly WasmTestCommandArguments _arguments;
+        private readonly Action<string> _processLogMessage;
+        private readonly ILogger _logger;
+        private readonly IEnumerable<string> _passThroughArguments;
+
+        // Messages from selenium prepend the url, and location where the message originated
+        // Eg. `foo` becomes `http://localhost:8000/xyz.js 0:12 "foo"
+        static readonly Lazy<Regex> s_consoleLogRegex = new Lazy<Regex>(() => new Regex(@"^\s*[a-z]*://[^\s]+\s+\d+:\d+\s+""(.*)""\s*$"));
```

Comment:
Does it make sense to precompile the regex maybe?

Reply:
Also not sure about the Laziness? Why not just get it ready before we start executing tests and measure time?

Reply:
Yeah, compiled sounds good. I'll remove the Lazy.